demo/src/execute.ts

```diff
@@ -35,28 +38,24 @@ export class AVRRunner {
     this.portC = new AVRIOPort(this.cpu, portCConfig);
     this.portD = new AVRIOPort(this.cpu, portDConfig);
     this.usart = new AVRUSART(this.cpu, usart0Config, this.speed);
+    this.taskScheduler = new MicroTaskScheduler();
+    this.taskScheduler.start();
   }
 
-  async execute(callback: (cpu: CPU) => void) {
-    this.stopped = false;
-    const workUnitCycles = 500000;
-    let nextTick = this.cpu.cycles + workUnitCycles;
-    for (;;) {
+  // CPU main loop
+  execute(callback: (cpu: CPU) => void) {
+    const nbCycles = this.cpu.cycles + this.workUnitCycles;
```

Comment:
what does the `nb` in `nbCycles` stand for? I couldn't figure it out while reading the code, it might make sense to choose a more meaningful name

Reply:
nb was for number, number of cycles. 

It is not clear indeed, i changed it to cyclesToRun, is it clearer ?